### DISCUSSION TITLE
Near field profile of multiple modes in a photonic crystal structure

### DISCUSSION THREAD

**Discussion by nkvij** [flexcompute/tidy3d#1082] [Q&A] [answered]
Hi,

I am studying photonic crystal structures that exhibit multiple high-Q modes. To find the frequency of the modes, I use a time-series monitor and the Resonance finder plug-in. This works out well.
Now, I want to look at the field profiles for each mode. For this, I use td.FieldMonitor, but the monitor only stores the mode profile at the frequency of the source dipole. How do I ensure I can see the mode profiles of the multiple modes?

**Comment by momchil-flex** [comment, marked as answer]
Put a FieldMonitor that records at the frequency of each of the modes, and use [apodization](https://docs.flexcompute.com/projects/tidy3d/en/latest/_autosummary/tidy3d.ApodizationSpec.html?highlight=apodizationspec) to exclude the time during which the source is active (note: the field normalization will be arbitrary). This should work well if your modes are reasonably well separated. However if you have modes that are close to degenerate, you may have to figure out a way to separate them additionally. This could be for example figuring out a source that would selectively excite one or the other (e.g. if one mode has a node where another has antinode).

**Reply by momchil-flex** [reply]
This is also discussed a bit in [this example](https://docs.flexcompute.com/projects/tidy3d/en/latest/notebooks/OptimizedL3.html?highlight=apodization#Initialize-monitors).
